h2o-algos/src/main/java/hex/DataInfo.java

```diff
@@ -1091,7 +1094,7 @@ public final int getCategoricalId(int cid, int val) {
     int [] offs = fullCatOffsets();
     int expandedVal = val + offs[cid];
     if(expandedVal >= offs[cid+1]) {  // previously unseen level
-      assert _valid : "Categorical value out of bounds, got " + val + ", next cat starts at " + fullCatOffsets()[cid + 1];
+      assert (isIWV && !_useAllFactorLevels) || _valid : "Categorical value out of bounds, got " + val + ", next cat starts at " + fullCatOffsets()[cid + 1];
```

Comment:
is this assert still needed?

Reply:
@michalk:  Yes, we still need to skip the original assert.  Because the use_all_factor_levels=False does not reduce the categorical levels like normal use.  For normal enum column, when use_all_factor_levels=False, the enum level just drop by one.  However, with interaction columns, the enum level is determined by number of enum level column1 (now is one less than the old level) times number of enum level column2(now is one less than the old level).  So, the assert still does not apply to interaction columns. However, the assert is still correct for regular enum columns and I would like to keep it there.